crates/virtio-queue/src/state.rs

```diff
@@ -149,6 +150,87 @@ impl QueueState {
             .map(Wrapping)
             .map_err(Error::GuestMemory)
     }
+
+    /// This is a test case that checks that an invalid avail_idx is not triggering a panic in
+    /// the queue implementation.
+    pub fn check_invalid_avail_idx<M: GuestMemory>(
```

Comment:
This function does not belong here and that's reason for the other problems you're observing. You should revert all changes that add `GuestMemoryMmap` and the lines that were deleting the `#[cfg(any(test, feature = "test-utils"))]`. For now, let's just move this to `mod tests` in `queue.rs`. To be able to re-use, you'll need to make `mod tests` public, and then import the function in queue_quard.rs similarly to how you're doing it now.

Reply:
Done